command/agent/config/config.go

```diff
@@ -112,11 +112,26 @@ func LoadConfig(path string, logger log.Logger) (*Config, error) {
 		return nil, errwrap.Wrapf("error parsing 'auto_auth': {{err}}", err)
 	}
 
+	err = parseListeners(&result, list)
+	if err != nil {
+		return nil, errwrap.Wrapf("error parsing 'listeners': {{err}}", err)
+	}
+
 	err = parseCache(&result, list)
 	if err != nil {
 		return nil, errwrap.Wrapf("error parsing 'cache':{{err}}", err)
 	}
 
+	if result.Cache != nil {
```

Comment:
Maybe move this logic into `parseCache`

Reply:
I thought about it, the reason I was reluctant is that then parseCache assumes a particular ordering of the calls in LoadConfig.

Reply:
I see, we can leave it as-is then.